`make release` fails on ubuntu 16.04

Similar to #37076 
this is to test the first contributor experience #36655 

`make release` fails on ubuntu 16.04, docker version 1.12.3:

```
...
+++ [1118 11:18:18] Generating bindata:
    /go/src/k8s.io/kubernetes/test/e2e/framework/gobindata_util.go
+++ [1118 11:18:19] Multiple platforms requested, but available 7G < threshold 11G, building platforms in serial
+++ [1118 11:18:19] Building the toolchain targets:
    k8s.io/kubernetes/hack/cmd/teststale
+++ [1118 11:18:19] Building go targets for linux/amd64:
    cmd/kubectl
    federation/cmd/kubefed
+++ [1118 11:18:37] Building go targets for linux/386:
    cmd/kubectl
    federation/cmd/kubefed
+++ [1118 11:20:05] Building go targets for linux/arm:
    cmd/kubectl
    federation/cmd/kubefed
+++ [1118 11:20:23] Building go targets for linux/arm64:
    cmd/kubectl
    federation/cmd/kubefed
+++ [1118 11:20:43] Building go targets for darwin/amd64:
    cmd/kubectl
    federation/cmd/kubefed
# runtime/cgo
aarch64-linux-gnu-gcc: error: unrecognized command line option '-m64'
Makefile:79: recipe for target 'all' failed
make[1]: *** [all] Error 1
Makefile:264: recipe for target 'cross' failed
make: *** [cross] Error 1
Makefile:248: recipe for target 'release' failed
make: *** [release] Error 1
$ docker version
Client:
 Version:      1.12.3
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   6b644ec
 Built:        Wed Oct 26 22:01:48 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.12.3
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   6b644ec
 Built:        Wed Oct 26 22:01:48 2016
 OS/Arch:      linux/amd64
ubuntu@gobuild:~/go/src/k8s.io/kubernetes$ cat /etc/os-release 
NAME="Ubuntu"
VERSION="16.04.1 LTS (Xenial Xerus)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 16.04.1 LTS"
VERSION_ID="16.04"
HOME_URL="http://www.ubuntu.com/"
SUPPORT_URL="http://help.ubuntu.com/"
BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
UBUNTU_CODENAME=xenial
ubuntu@gobuild:~/go/src/k8s.io/kubernetes$ uname -a
Linux gobuild 4.4.0-45-generic #66-Ubuntu SMP Wed Oct 19 14:12:37 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
```

tagir@ubuntu:~/gospace/kubernetes$ sudo make release
+++ [1123 13:33:46] Verifying Prerequisites....
+++ [1123 13:33:46] Building Docker image kube-build:build-911c53b32d-4-v1.7.3-0
+++ [1123 13:33:48] Syncing sources to container
+++ [1123 13:33:49] Running build command...
+++ [1123 13:34:16] Generating bindata:
    /go/src/k8s.io/kubernetes/test/e2e/framework/gobindata_util.go
+++ [1123 13:34:16] Multiple platforms requested, but available 3G < threshold 11G, building platforms in serial
+++ [1123 13:34:16] Building the toolchain targets:
    k8s.io/kubernetes/hack/cmd/teststale
+++ [1123 13:34:17] Building go targets for linux/amd64:
    cmd/kube-dns
    cmd/kube-proxy
    cmd/kube-apiserver
    cmd/kube-controller-manager
    cmd/kubelet
    cmd/kubeadm
    cmd/hyperkube
    cmd/kube-discovery
    plugin/cmd/kube-scheduler

+++ [1123 13:39:59] Building go targets for linux/arm:
    cmd/kube-dns
    cmd/kube-proxy
    cmd/kube-apiserver
    cmd/kube-controller-manager
    cmd/kubelet
    cmd/kubeadm
    cmd/hyperkube
    cmd/kube-discovery
    plugin/cmd/kube-scheduler
+++ [1123 13:46:56] Building go targets for linux/arm64:
    cmd/kube-dns
    cmd/kube-proxy
    cmd/kube-apiserver
    cmd/kube-controller-manager
    cmd/kubelet
    cmd/kubeadm
    cmd/hyperkube
    cmd/kube-discovery
    plugin/cmd/kube-scheduler
+++ [1123 13:55:44] Generating bindata:
    /go/src/k8s.io/kubernetes/test/e2e/framework/gobindata_util.go
+++ [1123 13:55:45] Multiple platforms requested, but available 3G < threshold 11G, building platforms in serial
+++ [1123 13:55:45] Building the toolchain targets:
    k8s.io/kubernetes/hack/cmd/teststale
+++ [1123 13:55:45] Building go targets for linux/amd64:
    cmd/kubectl
    federation/cmd/kubefed
+++ [1123 13:56:05] Building go targets for linux/386:
    cmd/kubectl
    federation/cmd/kubefed
+++ [1123 13:57:49] Building go targets for linux/arm:
    cmd/kubectl
    federation/cmd/kubefed
+++ [1123 13:58:10] Building go targets for linux/arm64:
    cmd/kubectl
    federation/cmd/kubefed
+++ [1123 13:58:31] Building go targets for darwin/amd64:
    cmd/kubectl
    federation/cmd/kubefed
# runtime/cgo
aarch64-linux-gnu-gcc: error: unrecognized command line option '-m64'
Makefile:79: recipe for target 'all' failed
make[1]: *** [all] Error 1
Makefile:264: recipe for target 'cross' failed
make: *** [cross] Error 1
make: *** [release] Error 1
tagir@ubuntu:~/gospace/kubernetes$

for me:

* `make` works (using your PR to fixes the gobindata dep) -local Go build-
* `make quick-release` works (docker build)
* `make clean` works if you have Docker running.

```
$ make clean
+++ [1129 14:00:12] Verifying Prerequisites....
+++ [1129 14:00:12] Using Docker for MacOS
Can't connect to 'docker' daemon.  please fix and retry.
...
```

* `make release` does not work:

```
...
    cmd/kube-controller-manager
    cmd/kubelet
    cmd/kubeadm
    cmd/hyperkube
    cmd/kube-discovery
    plugin/cmd/kube-scheduler
+++ [1129 10:03:26] Multiple platforms requested, but available 1G < threshold 11G, building platforms in serial
+++ [1129 10:03:26] Building the toolchain targets:
    k8s.io/kubernetes/hack/cmd/teststale
    k8s.io/kubernetes/vendor/github.com/jteeuwen/go-bindata/go-bindata
+++ [1129 10:03:27] Generating bindata:
    test/e2e/framework/gobindata_util.go
+++ [1129 10:03:28] Building go targets for linux/amd64:
    cmd/kubectl
    federation/cmd/kubefed
+++ [1129 10:03:53] Building go targets for linux/386:
    cmd/kubectl
    federation/cmd/kubefed
+++ [1129 10:06:08] Building go targets for linux/arm:
    cmd/kubectl
    federation/cmd/kubefed
+++ [1129 10:06:33] Building go targets for linux/arm64:
    cmd/kubectl
    federation/cmd/kubefed
+++ [1129 10:06:56] Building go targets for darwin/amd64:
    cmd/kubectl
    federation/cmd/kubefed
# runtime/cgo
aarch64-linux-gnu-gcc: error: unrecognized command line option '-m64'
Makefile:79: recipe for target 'all' failed
make[1]: *** [all] Error 1
Makefile:264: recipe for target 'cross' failed
make: *** [cross] Error 1
make: *** [release] Error 1
```

`V=4 make release` on OSX did not seem to give more verbosity.

'''
root@ubuntu:/home/tagir/gospace/kubernetes# V=4 make release
+++ [1129 12:31:46] Generating bindata:
    /go/src/k8s.io/kubernetes/test/e2e/framework/gobindata_util.go
+++ [1129 12:31:47] Multiple platforms requested, but available 6G < threshold 11G, building platforms in serial
+++ [1129 12:31:47] Building the toolchain targets:
    k8s.io/kubernetes/hack/cmd/teststale
+++ [1129 12:31:48] Building go targets for linux/amd64:
    cmd/libs/go2idl/deepcopy-gen
+++ [1129 12:31:50] Building go targets for linux/arm:
    cmd/libs/go2idl/deepcopy-gen
+++ [1129 12:31:54] Building go targets for linux/arm64:
    cmd/libs/go2idl/deepcopy-gen
+++ [1129 12:32:13] Generating bindata:
    /go/src/k8s.io/kubernetes/test/e2e/framework/gobindata_util.go
+++ [1129 12:32:14] Multiple platforms requested, but available 6G < threshold 11G, building platforms in serial
+++ [1129 12:32:14] Building the toolchain targets:
    k8s.io/kubernetes/hack/cmd/teststale
+++ [1129 12:32:14] Building go targets for linux/amd64:
    cmd/libs/go2idl/defaulter-gen
+++ [1129 12:32:15] Building go targets for linux/arm:
    cmd/libs/go2idl/defaulter-gen
+++ [1129 12:32:17] Building go targets for linux/arm64:
    cmd/libs/go2idl/defaulter-gen
+++ [1129 12:32:23] Generating bindata:
    /go/src/k8s.io/kubernetes/test/e2e/framework/gobindata_util.go
+++ [1129 12:32:24] Multiple platforms requested, but available 6G < threshold 11G, building platforms in serial
+++ [1129 12:32:24] Building the toolchain targets:
    k8s.io/kubernetes/hack/cmd/teststale
+++ [1129 12:32:24] Building go targets for linux/amd64:
    cmd/libs/go2idl/conversion-gen
+++ [1129 12:32:26] Building go targets for linux/arm:
    cmd/libs/go2idl/conversion-gen
+++ [1129 12:32:27] Building go targets for linux/arm64:
    cmd/libs/go2idl/conversion-gen
+++ [1129 12:32:36] Generating bindata:
    /go/src/k8s.io/kubernetes/test/e2e/framework/gobindata_util.go
+++ [1129 12:32:36] Multiple platforms requested, but available 6G < threshold 11G, building platforms in serial
+++ [1129 12:32:36] Building the toolchain targets:
    k8s.io/kubernetes/hack/cmd/teststale
+++ [1129 12:32:37] Building go targets for linux/amd64:
    cmd/libs/go2idl/openapi-gen
+++ [1129 12:32:42] Building go targets for linux/arm:
    cmd/libs/go2idl/openapi-gen
+++ [1129 12:32:48] Building go targets for linux/arm64:
    cmd/libs/go2idl/openapi-gen
+++ [1129 12:33:03] Generating bindata:
    /go/src/k8s.io/kubernetes/test/e2e/framework/gobindata_util.go
+++ [1129 12:33:04] Multiple platforms requested, but available 6G < threshold 11G, building platforms in serial
+++ [1129 12:33:04] Building the toolchain targets:
    k8s.io/kubernetes/hack/cmd/teststale
+++ [1129 12:33:04] Building go targets for linux/amd64:
    cmd/kube-dns
    cmd/kube-proxy
    cmd/kube-apiserver
    cmd/kube-controller-manager
    cmd/kubelet
    cmd/kubeadm
    cmd/hyperkube
    cmd/kube-discovery
    plugin/cmd/kube-scheduler
+++ [1129 12:42:46] Building go targets for linux/arm:
    cmd/kube-dns
    cmd/kube-proxy
    cmd/kube-apiserver
    cmd/kube-controller-manager
    cmd/kubelet
    cmd/kubeadm
    cmd/hyperkube
    cmd/kube-discovery
    plugin/cmd/kube-scheduler
+++ [1129 12:51:13] Building go targets for linux/arm64:
    cmd/kube-dns
    cmd/kube-proxy
    cmd/kube-apiserver
    cmd/kube-controller-manager
    cmd/kubelet
    cmd/kubeadm
    cmd/hyperkube
    cmd/kube-discovery
    plugin/cmd/kube-scheduler
+++ [1129 13:00:41] Generating bindata:
    /go/src/k8s.io/kubernetes/test/e2e/framework/gobindata_util.go
+++ [1129 13:00:42] Multiple platforms requested, but available 6G < threshold 11G, building platforms in serial
+++ [1129 13:00:42] Building the toolchain targets:
    k8s.io/kubernetes/hack/cmd/teststale
+++ [1129 13:00:42] Building go targets for linux/amd64:
    cmd/kubectl
    federation/cmd/kubefed
+++ [1129 13:01:03] Building go targets for linux/386:
    cmd/kubectl
    federation/cmd/kubefed
+++ [1129 13:03:37] Building go targets for linux/arm:
    cmd/kubectl
    federation/cmd/kubefed
+++ [1129 13:03:57] Building go targets for linux/arm64:
    cmd/kubectl
    federation/cmd/kubefed
+++ [1129 13:04:17] Building go targets for darwin/amd64:
    cmd/kubectl
    federation/cmd/kubefed
# runtime/cgo
aarch64-linux-gnu-gcc: error: unrecognized command line option '-m64'
Makefile:79: recipe for target 'all' failed
make[1]: *** [all] Error 1
Makefile:264: recipe for target 'cross' failed
make: *** [cross] Error 1
make: *** [release] Error 1
root@ubuntu:/home/tagir/gospace/kubernetes# 
'''

here we go:
```
root@ubuntu:/home/tagir/gospace/kubernetes# KUBE_VERBOSE=4 make release
+++ [1129 13:38:32] Verifying Prerequisites....
+++ [1129 13:38:32] Building Docker image kube-build:build-911c53b32d-4-v1.7.3-0
+++ [1129 13:38:45] Keeping container docker-nginx2
+++ [1129 13:38:45] Keeping container docker-nginx2
+++ [1129 13:38:45] Keeping container docker-nginx2
+++ [1129 13:38:45] Keeping image kube-build:build-911c53b32d-4-v1.7.3-0
+++ [1129 13:38:45] Creating data container kube-build-data-911c53b32d-4-v1.7.3-0
+++ [1129 13:38:46] Syncing sources to container
+++ [1129 13:38:46] Stopping any currently running rsyncd container
+++ [1129 13:38:46] Starting rsyncd container
+++ [1129 13:38:48] Running rsync
+++ [1129 13:38:53] Stopping any currently running rsyncd container
+++ [1129 13:38:53] Running build command...
+++ [1129 13:39:15] Generating bindata:
    /go/src/k8s.io/kubernetes/test/e2e/framework/gobindata_util.go
+++ [1129 13:39:15] Multiple platforms requested, but available 6G < threshold 11G, building platforms in serial
+++ [1129 13:39:15] Building the toolchain targets:
    k8s.io/kubernetes/hack/cmd/teststale
+++ [1129 13:39:16] Building go targets for linux/amd64:
    cmd/libs/go2idl/deepcopy-gen
+++ [1129 13:39:18] Building go targets for linux/arm:
    cmd/libs/go2idl/deepcopy-gen
+++ [1129 13:39:22] Building go targets for linux/arm64:
    cmd/libs/go2idl/deepcopy-gen
+++ [1129 13:39:39] Generating bindata:
    /go/src/k8s.io/kubernetes/test/e2e/framework/gobindata_util.go
+++ [1129 13:39:40] Multiple platforms requested, but available 6G < threshold 11G, building platforms in serial
+++ [1129 13:39:40] Building the toolchain targets:
    k8s.io/kubernetes/hack/cmd/teststale
+++ [1129 13:39:40] Building go targets for linux/amd64:
    cmd/libs/go2idl/defaulter-gen
+++ [1129 13:39:41] Building go targets for linux/arm:
    cmd/libs/go2idl/defaulter-gen
+++ [1129 13:39:43] Building go targets for linux/arm64:
    cmd/libs/go2idl/defaulter-gen
+++ [1129 13:39:50] Generating bindata:
    /go/src/k8s.io/kubernetes/test/e2e/framework/gobindata_util.go
+++ [1129 13:39:51] Multiple platforms requested, but available 6G < threshold 11G, building platforms in serial
+++ [1129 13:39:51] Building the toolchain targets:
    k8s.io/kubernetes/hack/cmd/teststale
+++ [1129 13:39:52] Building go targets for linux/amd64:
    cmd/libs/go2idl/conversion-gen
+++ [1129 13:39:53] Building go targets for linux/arm:
    cmd/libs/go2idl/conversion-gen
+++ [1129 13:39:54] Building go targets for linux/arm64:
    cmd/libs/go2idl/conversion-gen
+++ [1129 13:40:02] Generating bindata:
    /go/src/k8s.io/kubernetes/test/e2e/framework/gobindata_util.go
+++ [1129 13:40:03] Multiple platforms requested, but available 6G < threshold 11G, building platforms in serial
+++ [1129 13:40:03] Building the toolchain targets:
    k8s.io/kubernetes/hack/cmd/teststale
+++ [1129 13:40:03] Building go targets for linux/amd64:
    cmd/libs/go2idl/openapi-gen
+++ [1129 13:40:07] Building go targets for linux/arm:
    cmd/libs/go2idl/openapi-gen
+++ [1129 13:40:13] Building go targets for linux/arm64:
    cmd/libs/go2idl/openapi-gen
+++ [1129 13:40:27] Generating bindata:
    /go/src/k8s.io/kubernetes/test/e2e/framework/gobindata_util.go
+++ [1129 13:40:28] Multiple platforms requested, but available 6G < threshold 11G, building platforms in serial
+++ [1129 13:40:28] Building the toolchain targets:
    k8s.io/kubernetes/hack/cmd/teststale
+++ [1129 13:40:28] Building go targets for linux/amd64:
    cmd/kube-dns
    cmd/kube-proxy
    cmd/kube-apiserver
    cmd/kube-controller-manager
    cmd/kubelet
    cmd/kubeadm
    cmd/hyperkube
    cmd/kube-discovery
    plugin/cmd/kube-scheduler
+++ [1129 13:50:31] Building go targets for linux/arm:
    cmd/kube-dns
    cmd/kube-proxy
    cmd/kube-apiserver
    cmd/kube-controller-manager
    cmd/kubelet
    cmd/kubeadm
    cmd/hyperkube
    cmd/kube-discovery
    plugin/cmd/kube-scheduler
+++ [1129 14:02:14] Building go targets for linux/arm64:
    cmd/kube-dns
    cmd/kube-proxy
    cmd/kube-apiserver
    cmd/kube-controller-manager
    cmd/kubelet
    cmd/kubeadm
    cmd/hyperkube
    cmd/kube-discovery
    plugin/cmd/kube-scheduler
+++ [1129 14:15:24] Generating bindata:
    /go/src/k8s.io/kubernetes/test/e2e/framework/gobindata_util.go
+++ [1129 14:15:25] Multiple platforms requested, but available 6G < threshold 11G, building platforms in serial
+++ [1129 14:15:25] Building the toolchain targets:
    k8s.io/kubernetes/hack/cmd/teststale
+++ [1129 14:15:25] Building go targets for linux/amd64:
    cmd/kubectl
    federation/cmd/kubefed
+++ [1129 14:15:53] Building go targets for linux/386:
    cmd/kubectl
    federation/cmd/kubefed
+++ [1129 14:19:31] Building go targets for linux/arm:
    cmd/kubectl
    federation/cmd/kubefed
+++ [1129 14:19:52] Building go targets for linux/arm64:
    cmd/kubectl
    federation/cmd/kubefed
+++ [1129 14:20:20] Building go targets for darwin/amd64:
    cmd/kubectl
    federation/cmd/kubefed
# runtime/cgo
aarch64-linux-gnu-gcc: error: unrecognized command line option '-m64'
Makefile:79: recipe for target 'all' failed
make[1]: *** [all] Error 1
Makefile:264: recipe for target 'cross' failed
make: *** [cross] Error 1
!!! Error in build-tools/../build-tools/common.sh:580
  '"${docker_cmd[@]}" "${cmd[@]}"' exited with status 2
Call stack:
  1: build-tools/../build-tools/common.sh:580 kube::build::run_build_command_ex(...)
  2: build-tools/../build-tools/common.sh:515 kube::build::run_build_command(...)
  3: build-tools/release.sh:36 main(...)
Exiting with status 1
make: *** [release] Error 1
root@ubuntu:/home/tagir/gospace/kubernetes# 
```

I think it is almost there. Got much further in the build (4 GB of RAM).
Towards the end there is a failed test that makes the build fail:

```
ok  	k8s.io/kubernetes/pkg/kubelet/cadvisor	0.247s
ok  	k8s.io/kubernetes/pkg/kubelet/client	0.157s
ok  	k8s.io/kubernetes/pkg/kubelet/cm	0.204s
E1130 11:22:58.660773    8940 event.go:258] Could not construct reference to: '&v1.Pod{TypeMeta:unversioned.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:v1.ObjectMeta{Name:"foo", GenerateName:"", Namespace:"", SelfLink:"", UID:"foo", ResourceVersion:"", Generation:0, CreationTimestamp:unversioned.Time{Time:time.Time{sec:0, nsec:0, loc:(*time.Location)(nil)}}, DeletionTimestamp:(*unversioned.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string(nil), Annotations:map[string]string(nil), OwnerReferences:[]v1.OwnerReference(nil), Finalizers:[]string(nil), ClusterName:""}, Spec:v1.PodSpec{Volumes:[]v1.Volume(nil), InitContainers:[]v1.Container(nil), Containers:[]v1.Container{v1.Container{Name:"ctr", Image:"image", Command:[]string(nil), Args:[]string(nil), WorkingDir:"", Ports:[]v1.ContainerPort(nil), Env:[]v1.EnvVar(nil), Resources:v1.ResourceRequirements{Limits:v1.ResourceList(nil), Requests:v1.ResourceList(nil)}, VolumeMounts:[]v1.VolumeMount(nil), LivenessProbe:(*v1.Probe)(nil), ReadinessProbe:(*v1.Probe)(nil), Lifecycle:(*v1.Lifecycle)(nil), TerminationMessagePath:"", ImagePullPolicy:"IfNotPresent", SecurityContext:(*v1.SecurityContext)(0xc420276cf0), Stdin:false, StdinOnce:false, TTY:false}}, RestartPolicy:"Always", TerminationGracePeriodSeconds:(*int64)(nil), ActiveDeadlineSeconds:(*int64)(nil), DNSPolicy:"ClusterFirst", NodeSelector:map[string]string(nil), ServiceAccountName:"", DeprecatedServiceAccount:"", NodeName:"", HostNetwork:false, HostPID:false, HostIPC:false, SecurityContext:(*v1.PodSecurityContext)(nil), ImagePullSecrets:[]v1.LocalObjectReference(nil), Hostname:"", Subdomain:""}, Status:v1.PodStatus{Phase:"", Conditions:[]v1.PodCondition(nil), Message:"", Reason:"", HostIP:"", PodIP:"", StartTime:(*unversioned.Time)(nil), InitContainerStatuses:[]v1.ContainerStatus(nil), ContainerStatuses:[]v1.ContainerStatus(nil)}}' due to: 'selfLink was empty, can't make reference'. Will not report event: 'Warning' 'FailedValidation' 'Error validating pod foo.<empty-namespace> from test, ignoring: metadata.namespace: Required value'
E1130 11:22:58.662255    8940 event.go:258] Could not construct reference to: '&v1.Pod{TypeMeta:unversioned.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:v1.ObjectMeta{Name:"foo", GenerateName:"", Namespace:"", SelfLink:"", UID:"", ResourceVersion:"", Generation:0, CreationTimestamp:unversioned.Time{Time:time.Time{sec:0, nsec:0, loc:(*time.Location)(nil)}}, DeletionTimestamp:(*unversioned.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string(nil), Annotations:map[string]string(nil), OwnerReferences:[]v1.OwnerReference(nil), Finalizers:[]string(nil), ClusterName:""}, Spec:v1.PodSpec{Volumes:[]v1.Volume(nil), InitContainers:[]v1.Container(nil), Containers:[]v1.Container(nil), RestartPolicy:"", TerminationGracePeriodSeconds:(*int64)(nil), ActiveDeadlineSeconds:(*int64)(nil), DNSPolicy:"", NodeSelector:map[string]string(nil), ServiceAccountName:"", DeprecatedServiceAccount:"", NodeName:"", HostNetwork:false, HostPID:false, HostIPC:false, SecurityContext:(*v1.PodSecurityContext)(nil), ImagePullSecrets:[]v1.LocalObjectReference(nil), Hostname:"", Subdomain:""}, Status:v1.PodStatus{Phase:"", Conditions:[]v1.PodCondition(nil), Message:"", Reason:"", HostIP:"", PodIP:"", StartTime:(*unversioned.Time)(nil), InitContainerStatuses:[]v1.ContainerStatus(nil), ContainerStatuses:[]v1.ContainerStatus(nil)}}' due to: 'selfLink was empty, can't make reference'. Will not report event: 'Warning' 'FailedValidation' 'Error validating pod foo.<empty-namespace> from test, ignoring: [metadata.namespace: Required value, spec.containers: Required value, spec.restartPolicy: Required value, spec.dnsPolicy: Required value]'
E1130 11:22:58.667786    8940 file.go:72] unable to read config path "random_non_existent_path": path does not exist, ignoring
E1130 11:22:58.676588    8940 file.go:72] unable to read config path "/tmp.k8s/dir-test142653520/test_pod_config_0": path does not exist, ignoring
--- FAIL: TestReadPodsFromFileChanged (5.05s)
	file_linux_test.go:379: Simple pod: Expected update, timeout instead
FAIL
FAIL	k8s.io/kubernetes/pkg/kubelet/config	5.546s
ok  	k8s.io/kubernetes/pkg/kubelet/container	0.188s
```

This is probably a different bug, as your PR got passed the misconfigured compilation.

--- a/build-tools/common.sh
+++ b/build-tools/common.sh
@@ -560,6 +560,7 @@ function kube::build::run_build_command_ex() {
     --env "KUBE_FASTBUILD=${KUBE_FASTBUILD:-false}"
     --env "KUBE_BUILDER_OS=${OSTYPE:-notdetected}"
     --env "KUBE_BUILD_PPC64LE=${KUBE_BUILD_PPC64LE}"  # TODO(IBM): remove
+    --env "KUBE_VERBOSE=${KUBE_VERBOSE}"
   )
 
   # If we have stdin we can run interactive.  This allows things like 'shell.sh'

--- a/hack/lib/golang.sh
+++ b/hack/lib/golang.sh
@@ -219,6 +219,8 @@ kube::golang::set_platform_envs() {
   [[ -n ${1-} ]] || {
     kube::log::error_exit "!!! Internal error. No platform set in kube::golang::set_platform_envs"
   }
+  # make sure we have a clean slate first
+  kube::golang::unset_platform_envs
 
   export GOOS=${platform%/*}
   export GOARCH=${platform##*/}
@@ -456,6 +458,8 @@ kube::golang::build_binaries_for_platform() {
   local -a nonstatics=()
   local -a tests=()
 
+  V=2 kube::log::info "Env for ${platform}: GOOS=${GOOS-} GOARCH=${GOARCH-} GOROOT=${GOROOT-} CGO_ENABLED=${CGO_ENABLED-} CC=${CC-}"
+
   for binary in "${binaries[@]}"; do
 
     if [[ "${binary}" =~ ".test"$ ]]; then
@@ -685,8 +689,10 @@ kube::golang::build_binaries() {
     else
       for platform in "${platforms[@]}"; do
         kube::log::status "Building go targets for ${platform}:" "${targets[@]}"
-        kube::golang::set_platform_envs "${platform}"
-        kube::golang::build_binaries_for_platform ${platform} ${use_go_build:-}
+        (
+          kube::golang::set_platform_envs "${platform}"
+          kube::golang::build_binaries_for_platform ${platform} ${use_go_build:-}
+        )
       done
     fi
   )

